connman/src/connman.service.in

```diff
@@ -1,12 +1,15 @@
 [Unit]
 Description=Connection service
+DefaultDependencies=false
+Conflicts=shutdown.target
 Requires=@CONNMAN_SERVICE_REQUIRES@
 After=@CONNMAN_SERVICE_AFTER@ network-pre.target
-Before=network.target
+Before=network.target multi-user.target shutdown.target
 Wants=network.target
 
 [Service]
-Type=notify
+Type=dbus
```

Comment:
If unit becomes ready based on claiming dbus name, then it probably should not do sd_notify(), i.e. "--systemd" should be removed from ExecStart, or?

Then again, when should connman be considered "ready": Midway through initialization (what one presumably gets with type=dbus given where the name reservation happens) or when it enters mainloop (where sd_notify() gets called)? Having 10x delta between those two makes me wonder if dbus name is claimed too early / how long is that delta in absolute terms / is it long enough to cause timeouts for incoming method calls?

Reply:
Well we've been using that "nofity" for a really long time without any proper reason as upstream has defined the "dbus" type. I'd assume that this is how it should be.

Reply:
Ack. But then that --systemd arg is not needed. Probably does not hurt as it comes later / maybe systemd igores it due to type=dbus -> I'll leave it up to you to remove/leave it.

Reply:
And when I really looked at it... [upstream does not use that notifying of systemd](https://git.kernel.org/pub/scm/network/connman/connman.git/tree/src/main.c#n965) nor it has that in use anywhere. Again something old and specific to our fork. Well, lets not remove that just now but maybe we need to sync that later on with upstream as that does implement the systemd resolved support.

Reply:
We could revert e59306cc61ec1807cbd7fc1a2838ecd6764bf120 while at it. But then again, if someone decides to enable notify type it does not work properly, however, that would be an invalid configuration of the service after the revert..

Reply:
> > Ack. But then that --systemd arg is not needed. Probably does not hurt as it comes later / maybe systemd igores it due to type=dbus -> I'll leave it up to you to remove/leave it.
> 
> Nope, we cannot remove that `--systemd` as the login manager is relying on that setting to enable the systemd login manager.

Umm, yeah, my bad, that option has no actual effect on the login manager. There is a config option for it. My memory did not serve well.